logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueReader.java

```diff
@@ -393,5 +405,9 @@ public void close() throws IOException {
             currentReader.close();
         }
         this.watchService.close();
+
+        if (this.cleanConsumed) {
```

Comment:
Maybe put this in a `finally` block to ensure cleanup even if closing reader or watchService fails

Reply:
Fixed by 158aa2efce0316a85f67604a983f8316fab9e01a